src/jobs/jobs.ts

```diff
@@ -1,18 +1,41 @@
 import { ALBResult } from 'aws-lambda';
 import { NEWSLETTERS_BUCKET_NAME } from '../constants';
 import { s3upload } from '../lib/s3Upload';
-import { getEmailNewsletters } from './newsletters';
+import {
+	getEmailNewsletters,
+	getEmailNewslettersIncludingCancelled,
+} from './newsletters';
+
+const getAndUploadNewsletters = async (
```

Comment:
I think it's a little odd to upload a duplicated set of newsletters to s3. I appreciate this has the advantage of making the query faster, but it also means we duplicate a lot of the data, using more storage and no longer have one source of truth in s3 (what happens if these files start getting out of sync? etc.) 

I think the better approach here is to always upload all newsletters (including cancelled) and filter the list on the server based on the cancelled attribute. 

This also allows us to scale the solution for other attributes we might want to filter by (I don't think we should create a new s3 file every time we want a different subset of newsletters, again because keeping that data in sync becomes very tricky)

Reply:
I think I generally agree with this point. However, the architecture of the Newsletters API doesn't include a "server" which responds to requests. Instead, the domain points to Fastly which serves data from this S3 bucket directly.

I'm wondering if it is worth rearchitecting the infrastructure so that we could apply filtering dynamically. This might be in the form of a server, or possibly using something like Fastly's Compute@Edge to filter the JSON when the request is made.